go/pkg/client/cas.go

```diff
@@ -377,12 +380,9 @@ func (c *Client) BatchDownloadBlobs(ctx context.Context, dgs []digest.Digest) (m
 func (c *Client) makeBatches(dgs []digest.Digest) [][]digest.Digest {
 	var batches [][]digest.Digest
 	log.V(2).Infof("Batching %d digests", len(dgs))
-	sort.Slice(dgs, func(i, j int) bool {
-		return dgs[i].Size < dgs[j].Size
-	})
 	for len(dgs) > 0 {
-		batch := []digest.Digest{dgs[len(dgs)-1]}
```

Comment:
I think this, using back instead of front, optimization is for "large and small blobs hitting max exactly" test case, but that test case is bit artifical rather than pratical?

If reclient really needs this optimization, I will keep this optimization.

Reply:
The way I'd like to see this change landing is: rename the makeBatches function to makeBatchesOptimizeSize, and add another function named makeBatchesOptimizeLocality. Use makeBatchesOptimizeLocality for downloads, and probably on Windows to start, until there are some load test results to show benefits for Linux as well.
Keep the old tests (limit them by platform, if needed), and add Windows-only new test cases for the new functionality.

Reply:
Due to passing array of digest is not sufficient for `makeBatchesOptimizeLocality`, I changed to pass additional flag to `makeBatches`. Also expose that as behavior flag and disabled locality ordering by default rather than setting it based on OS. How do you think this?

Reply:
Better! I am in the process of a major rewrite to the upload/download logic (I will have a single uploader/downloader to manage uploads/downloads from many simultaneous actions), so I suspect I'll refactor this code, but I'll keep and respect your UtilizeLocality option. And if we want to make it the default later, we could.